tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -714,6 +715,47 @@ public:
    void AcceptVisitor(Detail::RFieldVisitor &visitor) const final;
 };
 
+/// The generic field an std::bitset<N>. All compilers we care about store the bits in an array of unsigned long.
```

Comment:
```suggestion
/// The generic field for an std::bitset<N>. All compilers we care about store the bits in an array of unsigned long.
```

Other than that, according to https://en.cppreference.com/w/cpp/utility/bitset/bitset, the internal value representation since C++11 is `unsigned long long`; however, `sizeof(unsigned long) == sizeof(unsigned long long)` for most (all?) target architectures.

Reply:
Good point, thanks for spotting that! Windows is an exception, I think, with `sizeof(unsigned long) == 4`

Reply:
Oops, my bad!  :slightly_frowning_face: 

That was only an interface change since C++11, but still, at least libstdc++ (GCC) uses `unsigned long` as seen below (`_M_w` is the backing array for the bitset).
```c++
      typedef unsigned long _WordT;

      /// 0 is the least significant word.
      _WordT 		_M_w[_Nw];
```
Anyways, I think we should double-check the internal representation in libc++ (llvm) and MSVC.

Reply:
I checked on Godbolt, `unsigned long` seems good.

Reply:
Is there a way for us to assert on the assumption (i.e. if there is a new compiler/platform, will we notice if the assumption is wrong)?

Reply:
Assertions on the `sizeof()` of the data type may help in that regard, e.g. https://github.com/jblomer/root/blob/e870de61a05f8fed67f32b5b8de95e8d62942fc1/tree/ntuple/v7/test/ntuple_types.cxx#L470, I think.